sisl/physics/electron.py

```diff
@@ -808,6 +816,14 @@ def _berry(eigenstates):
                 prd = _process(prd, prev.inner(second, False))
                 prev = second
 
+            # Insert Bloch phase for 1D integral?
+            if method == 'zak':
+                g = contour.parent.geometry
+                axis = contour.k[1] - contour.k[0]
+                axis /= np.linalg.norm(axis)
```

Comment:
You don't have to do this, I will change it after merging, but `np.linalg.norm` is horrendously slow, therefore I always do: `(axis ** 2).sum() ** 0.5`.

The `norm` function was really a performance bottleneck in `Hamiltonian.construct` for very large systems.

Reply:
OK, good to know! But axis is just a 3-element vector so performance is probably not affected by this.

Reply:
Agreed, no need to change (I'll do it after merge)

Reply:
I will change this too